pkg/gardenlet/controller/shoot/shoot_control_reconcile.go

```diff
@@ -74,6 +78,14 @@ func (r *shootReconciler) runReconcileShootFlow(ctx context.Context, o *operatio
 				return retryutils.Ok()
 			})
 		}),
+		errors.ToExecute("Check whether control plane exists and is succeeded", func() error {
+			cp := &extensionsv1alpha1.ControlPlane{}
+			if err := o.K8sSeedClient.Client().Get(ctx, kutil.Key(o.Shoot.SeedNamespace, o.Shoot.GetInfo().Name), cp); client.IgnoreNotFound(err) != nil {
+				return err
+			}
+			controlPlaneExistsAndSucceeded = !apierrors.IsNotFound(err) && cp.Status.LastOperation != nil && cp.Status.LastOperation.State == gardencorev1beta1.LastOperationStateSucceeded
+			return nil
```

Comment:
At first look with @oliver-goetz we had small confusion with the usages of the multiple not found checks. Maybe a more verbose way that is one idea easier to comprehend at first look:

```suggestion
			cp := &extensionsv1alpha1.ControlPlane{}
			if err := o.K8sSeedClient.Client().Get(ctx, kutil.Key(o.Shoot.SeedNamespace, o.Shoot.GetInfo().Name), cp); err != nil {
				if apierrors.IsNotFound(err) {
					controlPlaneExistsAndSucceeded = false
					return nil
				}

				return err
			}

			controlPlaneExistsAndSucceeded = cp.Status.LastOperation != nil && cp.Status.LastOperation.State == gardencorev1beta1.LastOperationStateSucceeded
			return nil
```